jupyter_sphinx/execute.py

```diff
@@ -443,14 +503,43 @@ def cell_output_to_nodes(cell, data_priority, dir):
     return to_add
 
 
-def attach_outputs(output_nodes, node):
-    if node.attributes['hide_code']:
-        node.children = []
-    if not node.attributes['hide_output']:
-        if node.attributes['code_below']:
-            node.children = output_nodes + node.children
-        else:
-            node.children = node.children + output_nodes
+def attach_outputs(output_nodes, node, thebe_config):
+    if thebe_config and not node.attributes['no_thebelab']:
+        code_class = 'thebelab-code'
+        if node.attributes['hide_code']:
+            code_class += ' thebelab-hidden'
+
+        code = node.astext()
+
+        node.children = [docutils.nodes.raw(
+            text='<pre class="{code_class}" data-executable="true" data-language="python">' \
+                '{data}' \
+                '</pre>'.format(data=code, code_class=code_class),
+            format='html',
+        )]
+
+        if not node.attributes['hide_output']:
+            # We ignore the code_below attribute since this is not supported with thebelab
+            node.children.append(docutils.nodes.raw(
+                text='<div class="thebelab-output" data-output="true">',
+                format='html',
+            ))
```

Comment:
same as above; how does this degrade when not using an HTML writer?

Reply:
and shouldn't we add children to this `div` node, rather than having the opening and closing tags as separate nodes?